src/features/workspace/components/workspace-models-dropdown.tsx

```diff
@@ -0,0 +1,160 @@
+} from '@/api/generated'
+import {
+  DialogTrigger,
+  Button,
+  Popover,
+  SearchField,
+  ListBox,
+  Input,
+  OptionRenderer,
+  OptionsSchema,
+} from '@stacklok/ui-kit'
+import { ChevronDown, SearchMd } from '@untitled-ui/icons-react'
+import { useState } from 'react'
+
+type Props = {
+  rule: MuxRule & { id: string }
+  isArchived: boolean
+  models: V1ListAllModelsForAllProvidersResponse
+  onChange: ({
+    model,
+    provider_id,
+  }: {
+    model: string
+    provider_id: string
+  }) => void
+}
+
+function groupModelsByProviderName(
+  models: ModelByProvider[]
+): OptionsSchema<'listbox', string>[] {
+  return models.reduce<OptionsSchema<'listbox', string>[]>(
+    (groupedProviders, item) => {
+      const providerData = groupedProviders.find(
+        (group) => group.id === item.provider_name
+      )
+      if (!providerData) {
+        groupedProviders.push({
+          id: item.provider_name,
+          items: [],
+          textValue: item.provider_name,
+        })
+      }
+
+      ;(providerData?.items ?? []).push({
+        id: `${item.provider_id}_${item.name}`,
+        textValue: item.name,
+      })
+
+      return groupedProviders
+    },
+    []
+  )
+}
+
+function filterModels({
+  groupedModels,
+  searchItem,
+}: {
+  searchItem: string
+  groupedModels: OptionsSchema<'listbox', string>[]
+}) {
+  const test = groupedModels
+    .map((modelData) => {
+      if (!searchItem) return modelData
+      const filteredModels = modelData.items?.filter((item) => {
+        return item.textValue.includes(searchItem)
+      })
+
+      const data = {
+        ...modelData,
+        items: filteredModels,
+      }
+      return data
+    })
+    .filter((item) => (item.items ? item.items.length > 0 : false))
+
+  return test
+}
+
+export function WorkspaceModelsDropdown({
+  rule,
+  isArchived,
+  models = [],
+  onChange,
+}: Props) {
+  const [isOpen, setIsOpen] = useState(false)
+  const [searchItem, setSearchItem] = useState('')
+  const groupedModels = groupModelsByProviderName(models)
+  console.log({ groupedModels, rule: rule })
+  const currentProvider = models.find((p) => p.provider_id === rule.provider_id)
+  const currentModel =
+    currentProvider && rule.model
+      ? `${currentProvider?.provider_name}/${rule.model}`
+      : ''
+  const selectedKey = `${rule.provider_id}_${rule.model}`
+
+  return (
+    <div className="flex w-full">
+      <DialogTrigger isOpen={isOpen} onOpenChange={(test) => setIsOpen(test)}>
+        <Button
+          variant="secondary"
+          isDisabled={isArchived}
+          data-testid="workspace-models-dropdown"
+          className="flex w-full cursor-pointer justify-between border-gray-400 bg-gray-25
+            font-normal shadow-none"
+        >
+          <span>{currentModel || 'Select a model'}</span>
+          <ChevronDown className="shrink-0" />
+        </Button>
+
+        <Popover className="w-[32rem] p-3" placement="top end">
```

Comment:
maybe the popover content could/should be extracted to a separate component? 

I would even say the same for the trigger button, I like componetns to be pretty minimal

Reply:
My intention is to do this for workspace selection too, so yep cause they are pretty similar, but I would do it in a separate pr